pkg/api/annotations.go

```diff
@@ -64,22 +64,7 @@ func (hs *HTTPServer) PostAnnotation(c *models.ReqContext) response.Response {
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
 
-	var canSave bool
-	var err error
-	if cmd.DashboardId != 0 {
-		canSave, err = canSaveDashboardAnnotation(c, cmd.DashboardId)
```

Comment:
Here we should also check if the user has the permissions to create dashboard annotations. I've added this check [here](https://github.com/grafana/grafana/pull/47882/files#diff-0e170c34bf6bc7d5ebba5c28db138f77234a683021832a0a10d9eff8808f122cR449-R454).